idk/kafka/cmd_test.go

```diff
@@ -682,3 +702,122 @@ func tDoHTTPPost(t *testing.T, url, contentType, body string) string {
 
 	return string(bod)
 }
+
+/*
+In some cases, when we see max.poll.interval.ms exceeded on the kafka consumer,
+we also see the consumer hang when it's trying to close properly. See SUP-288.
+This test:
+ 1. configures the consumer such that it will hang if max.poll.interval.ms is exceeded.
+ 2. checks that
+    a. the consumer did leave the group due to exceeding max.poll.interval.ms
+    b. we close the consumer and log that we weren't able to confirm it closed properly
+
+It could be possible this test fails even though there isn't necessarily an issue. This
+would be the case if a fix comes along to the kafka-go lib that prevents the consumer
+from hanging during the call it's consumer.Close() func.
+*/
+func TestCloseTimeout(t *testing.T) {
+
+	//define some vars
+	source := "bank"
+	target := "kafka"
+	now := time.Now().UnixNano()
+	index := fmt.Sprintf("close_timeout_%d", now)
+	topic := fmt.Sprintf("close_timeout_%d", now)
+	subject := fmt.Sprintf("close_timeout_%d_subject", now)
+	log_path := fmt.Sprintf("./consumer_%d.log", now)
+	genNumRecords := 5000
+
+	// configure the consumer
+	consumer, err := NewMain()
+	if err != nil {
+		t.Fatalf("issues creating a consumer")
+	}
+	configureTestFlags(consumer)
+	consumer.Topics = []string{topic}
+	consumer.BatchSize = 1000
+	consumer.IDField = "user_id"
+	consumer.KafkaGroupInstanceId = "some_id"
+	consumer.ConsumerCloseTimeout = 3
+	consumer.KafkaMaxPollInterval = "15000"
+	consumer.MaxMsgs = 10000
+	consumer.Index = index
+	consumer.KafkaSessionTimeout = "10000"
+	consumer.LogPath = log_path
+
+	// run datagen and capture stdout and stderr in case of error
+	var datagen_0_stdout, datagen_0_stderr bytes.Buffer
+	cmdString := fmt.Sprintf("go run ../cmd/datagen/main.go --source %s --target %s --kafka.topic %s --kafka.subject %s --end-at %d --kafka.confluent-command.schema-registry-url %s --kafka.confluent-command.kafka-bootstrap-servers %s", source, target, topic, subject, genNumRecords, consumer.SchemaRegistryURL, consumer.KafkaBootstrapServers[0])
+	cmd := exec.Command("bash", "-c", cmdString)
+	cmd.Stdout = &datagen_0_stdout
+	cmd.Stderr = &datagen_0_stderr
+	err = cmd.Run()
+	if err != nil {
+		t.Errorf("stdout: %s", datagen_0_stdout.String())
+		t.Errorf("stderr: %s", datagen_0_stderr.String())
+		t.Fatalf("issue generating records for kafka: %s", err)
+	}
+
+	// run the consumer
+	consumerError := make(chan error)
+	go func() {
+		consumerErr := consumer.Run()
+		consumerError <- consumerErr
+	}()
+
+	// wait until all 5000 messages to be consumed
+	complete := false
+	start := time.Now()
+
+	for !complete {
+		buf, _ := os.ReadFile(log_path)
+		s := string(buf)
+		if strings.Contains(s, "records processed 0-> (5000)") {
```

Comment:
So, what happens if someone decides this is taking too long and edits `genNumRecords` to be 2000? This presumably never happens. Can we make this contingent on that behavior? Or is there some other way we can have this actually directly relate to the consumer's behavior, like having the consumer close a channel when it's done?

Reply:
So.. we need the consumer to stay alive so that it can, later in the test, timeout because it doesn't poll for messages. But I can format the string to include genNumRecords and I think that is a good call out!